### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by mikedewar**
don't be shy; don't merge without reading!

### CODE REVIEW

File: core/block.go [hunk 1]
```diff
@@ -29,6 +29,13 @@ func NewBlock(s Spec) *Block {
 		})
 	}
 
+	var sourceType SourceType
+	if s.Source != 0 {
+		sourceType = s.Source
+	} else {
+		sourceType = NONE
```
Comment (mikedewar) [inline core/block.go]: does NONE == 0 anyway? do I need this if statement?

Comment (nikhan) [inline core/block.go]: I believe NONE is 0, but semantically we are thinking NONE means "no source"

**Comment by nikhan** [comment]
I believe some types need to be changed in /core/server for this merge to compile?

File: core/block.go (hunk 1 above)
Comment (mikedewar) [inline core/block.go]: I think probably this block just needs to be

```sourceType = s.Source```

then? Or, even, just put s.Source in the literal below?

Comment (nikhan) [inline core/block.go]: yeah

**Comment by mikedewar** [comment]
Yep yep.

**Comment by mikedewar** [comment]
ok server compiles now. shall we move `/server` up a level so it's on the same level as `/core`?

**Comment by nikhan** [comment]
cool thanks, let's leave moving /server for another time
:cow: :dollar:

**Pull request merged**